MassRegistrationOfSAPSystems/RegisterSIDs.ps1

```diff
@@ -65,57 +107,76 @@ foreach($line in $file)
         }
         else
         {
-            Write-Host "Reached maximum parallel jobs, waiting for 30 seconds"
+            Write-Host "Reached maximum parallel jobs, waiting for $MonitoringIntervalInSeconds seconds"
             Start-Sleep -Seconds $MonitoringIntervalInSeconds
         }
     }
 
-    # Storing Job Definitions
+    # Storing Job Definitions to monitor the status of the jobs
     $Jobdefs += New-Object PSCustomObject -Property @{JobName = $random; SID = $SID; State = "Running"}
 }
 
 Write-Host "Waiting for all jobs to complete"
 
+# Declaring varaible as array for storing completed jobs
 $Completed = [System.Collections.ArrayList]::new()
 
 while($Completed.Count -ne ($Jobdefs | Measure-Object).Count)
 {
+    # Waiting for MonitoringIntervalInSeconds seconds before checking the status of the jobs
     Start-Sleep -Seconds $MonitoringIntervalInSeconds
 
     Write-Host "$($Completed.Count) Jobs completed out of $(($Jobdefs | Measure-Object).Count)"
 
+    # Iterating through each job definition
     foreach($Jobdef in $Jobdefs)
     {
 
+        # Checking if the job is already completed, if yes then skip the loop
         if($Completed.Contains($Jobdef.JobName))
         {
             continue
         }
-    
+        
+        # Checking the status of the job
         $Job = Get-Job -Name $($Jobdef.JobName)
 
+        # Checking if the job is not running
         if($Job.State -ne "Running")
         {
+            # Getting the output of the job
             $Out = Receive-Job -Name $($Jobdef.JobName) | ConvertFrom-Json
-
-            Write-Host "The registration for $($Jobdef.SID) completed with provisioning state as $($Out.properties.ProvisioningState)"
-
+            
+            # Adding the job to completed jobs list doesn't matter if it is succeeded or failed so that it is not checked again
             $null = $Completed.Add($Jobdef.JobName)
             
+            # Fetching the record from the input file for the current job
             $filerec = $file | Where-Object {$_.SID -eq $Jobdef.SID}
-
-            $filerec.State = $Out.properties.ProvisioningState
             
+            # Checking if the job is succeeded or failed and updating the status in record
+            if($Out.properties.ProvisioningState -eq "Succeeded")
+            {
+                $filerec.State = "Succeeded"
+            }
+            else
+            {
+                $filerec.State = "Failed"
+            }
+            
+            Write-Host "The registration for $($Jobdef.SID) completed with provisioning state as $($filerec.State)"
+            
+            # Updating the details of the job in a variable
             $Jobrec = $Jobdefs | Where-Object {$_.SID -eq $Jobdef.SID}
             
+            # Updating the status of the job in the variable
```

Comment:
Updating the status of the job in a variable

used 'a' instead of 'the' to keep it consistent with the other sentence too

Reply:
Completed the necessary modifications.